src/index.ts

```diff
@@ -182,37 +178,40 @@ export = function plugin(
       }
     },
 
-    generateBundle(outputOptions, _, isWrite) {
+    async generateBundle(outputOptions, _, isWrite) {
       const { styles } = pluginState;
       const { output, insert } = pluginOptions;
 
       if (!isWrite || (!insert && (!styles.length || output === false))) {
-        return Promise.resolve();
+        return;
       }
 
       const css = styles.map((style) => style.content).join('');
 
       if (typeof output === 'string') {
-        return fs.promises
-          .mkdir(path.dirname(output as string), { recursive: true })
-          .then(() => fs.promises.writeFile(output as string, css));
-      } else if (typeof output === 'function') {
-        return Promise.resolve(
-          (output as RollupPluginSassOutputFn)(css, styles),
-        );
-      } else if (!insert && outputOptions.file && output === true) {
+        await fs.promises.mkdir(path.dirname(output), { recursive: true });
+        await fs.promises.writeFile(output, css);
+
+        return;
+      }
+
+      if (typeof output === 'function') {
+        output(css, styles);
+        return;
+      }
+
+      if (!insert && outputOptions.file && output === true) {
         let dest = outputOptions.file;
 
         if (dest.endsWith('.js') || dest.endsWith('.ts')) {
           dest = dest.slice(0, -3);
         }
         dest = `${dest}.css`;
-        return fs.promises
-          .mkdir(path.dirname(dest), { recursive: true })
-          .then(() => fs.promises.writeFile(dest, css));
-      }
 
-      return Promise.resolve(css);
```

Comment:
[generateBundle is a `void` function](https://rollupjs.org/plugin-development/#generatebundle) so this return is not required. 
A promise will be returned always since it is an `async` function now.